src/phases/query/CollScanOnMixedDataTypes.yml

```diff
@@ -0,0 +1,275 @@
+  MaxPhases: &MaxPhases 12
+
+  # Generates a random numerical string of 16 characters.
+  RandomShortStringValue: &RandomShortStringValue {^FastRandomString: {
+    length: 16, alphabet: "0123456789"
+  }}
+
+  # Generates a random numerical string of 200 characters.
+  RandomMediumStringValue: &RandomMediumStringValue {^FastRandomString: {
+    length: 200, alphabet: "0123456789"
+  }}
+
+  # Generates a random numerical string of 50-250 characters.
+  RandomLongStringValue: &RandomLongStringValue {^FastRandomString: {
+    length: 1000, alphabet: "0123456789"
+  }}
+
+  RandomDateValue: &RandomDateValue {^RandomDate: {
+    min: "2021-01-01T00:00:02.000Z", max: "2054-01-01T00:00:10.000Z"
+  }}
+
+  RandomIntValue: &RandomIntValue {^RandomInt: {
+    min: 0, max: 20
+  }}
+
+  RandomDoubleValue: &RandomDoubleValue {^RandomDouble: {
+    min: 0, max: 10000
+  }}
+
+  RandomDecimalValue: &RandomDecimalValue {^ConvertToDecimal: {
+    from: *RandomDoubleValue
+  }}
+  
+  RandomArrayValue: &RandomArrayValue [
+    {^Array: {of: [[*RandomIntValue]], number: *RandomIntValue}}
+  ]
+
+  RandomObjectId: &RandomObjectId {^ObjectId: {
+    ^RandomString: {length: 24, alphabet: "0123456789ABCDEF"
+  }}}
+
+
+Actors:
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: *Collection
+        Operations:
+        - OperationName: drop
+
+- Name: InsertData
+  Type: Loader
+  Threads: 4
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document: {
+          numInt: *RandomIntValue,
+          numDouble: *RandomDoubleValue,
+          numDecimal: *RandomDecimalValue,
+          strS: *RandomShortStringValue,
+          strM: *RandomMediumStringValue,
+          strL: *RandomLongStringValue,
+          arr: *RandomArrayValue,
+          date: *RandomDateValue,
+          objID: *RandomObjectId,
+          obj: {
+            nested: {
+              strM: *RandomMediumStringValue,
+            }
+          }
+        }
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+
+- Name: CollScanQueryOnInt
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [3]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: *Repeat
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: { "numInt": {$gt: 10} }
```

Comment:
Are all these queries meant to have approximately the same cardinality? If not, how were the constants selected for each individual query?

Reply:
Good point. Basically I just chose the value in the middle of the corresponding range. It should have "approximately" the same cardinality, but I'm gonna use the min/max value of the range to maximize the cardinality so it should be stable at least.